### PR TITLE
Adds system testing workflow to github actions

### PR DESCRIPTION
**Pull request by ManavalanG**
# Pull request

* Adds a github actions workflow to perform system testing. 
* It [currently takes ~1.5-1.75hrs](https://github.com/uab-cgds-worthey/quac/actions/workflows/system_testing.yml) as it runs 4 system testing jobs in serial fashion.
* Refer #74 to  see discussions on optimization to make the workflow to run faster

Please fill in the checklist below and comment as needed.

- [x] Was code modified? Briefly describe. _Yes. System testing github actions workflow was added._
- [x] Was documentation modified? Briefly describe. _No._
- [x] Is this a bug-fix? Briefly describe. _No._
- [x] Is this a feature addition? Briefly describe. _Yes. See description above._
- [x] Did you modify QuaC-Watch config file? If so, did you modify multiqc template
  `configs/multiqc_config_template.jinja2` and script `src/quac_watch/create_mutliqc_configs.py` as necessary? _No._
- [x] Did you perform system-level testing manually as described in master readme doc? Did it pass completely? If not why? _No, as this is now done using github actions_
- [x] Updated `Changelog.md` file with change logs in recommended format? _Yes._

## Anything else reviewer should know?

No.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -8,7 +8,6 @@ Please fill in the checklist below and comment as needed.
 - [ ] Is this a feature addition? Briefly describe.
 - [ ] Did you modify QuaC-Watch config file? If so, did you modify multiqc template
   `configs/multiqc_config_template.jinja2` and script `src/quac_watch/create_mutliqc_configs.py` as necessary?
-- [ ] Did you perform system-level testing manually as described in master readme doc? Did it pass completely? If not why?
```
Comment (ManavalanG) [inline .github/pull_request_template.md:11]: Github actions now can take over system testing, but one thing it still doesn't test is whether quac works as expected sending jobs to slurm cluster. Would you call this an acceptable gap? I'm divided but leaning towards calling it as acceptable gap.

Comment (ManavalanG) [reply .github/pull_request_template.md:15]: In light of what we have seen as part of JOSS reviews, I now feel we should include a step to run a system testing job that utilizes sending jobs to slurm cluster.

**Review by wilkb777 — APPROVED** [review]
LGTM

**Comment by ManavalanG** [comment]
Suddenly github actions for system testing keep failing, even though no changes have been made to concerning source code. It is unclear why this is happening :(

I got access to to [larger runners](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners) but learnt later that [they are not free to use](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates). Relevant snippet:

```
* For larger runners, there is no additional cost for configurations that assign public static IP addresses to a larger runner. For more information on larger runners, see "Using larger runners."
* Entitlement minutes cannot be used for larger runners.
* The larger runners are not free for public repositories.
[4:56](https://cgds-uab.slack.com/archives/GTSEU473K/p1686866168293039)
``` 

I will get around to debugging the issue some day :(

**Comment by ManavalanG** [comment]
@wilkb777 Github actions workflow for system testing began failing despite no significant code changes, and the error had to do with lack of storage space. It is unclear why it worked before but started failing of late. I reran a workflow that had succeeded before, and it still errored out. After a series of debugging efforts (documented in #78), storage in the runners were freed first before running the github actions workflow. This resulted in [resolving the issue](https://github.com/uab-cgds-worthey/quac/actions/runs/5325997898/jobs/9647439319#logs).

So overall, I added a step in github actions workflow for system testing to free up disk storage space. Could you review the changes please? :)

File: .github/pull_request_template.md (hunk 1 above)
Comment (wilkb777) [reply .github/pull_request_template.md:15]: Hmmm that would be a bit more work but would truly be a full system test. Some of the issues noted have been due to the different environments set ups at different locations so this testing wouldn't alleviate that possibility. However, having a test her where jobs are submitted and handled through slurm would be more complete. I'm not sure though how much we gain from this kind of test though.

Comment (ManavalanG) [reply .github/pull_request_template.md:15]: To add to our earlier sidebar on this topic, the line of interest has been modified to this:

```
Did you perform system-level testing manually, using `--subtasks_slurm` flag, as described in the [documentation](https://quac.readthedocs.io/en/stable/system_testing/)? Did it pass completely? If not why?
```

Unsure why the changes made are not showing up in PR view above.

**Pull request merged**